### PR TITLE
replicator client reconnect more than once

### LINKED ISSUE #1768: BinlogClient tries reconnect only once | TimeoutException
Sometimes we have a connection issues and we need to binlog client reconnect more then once.

steps to reproduce:
1.Run maxwel, params:  --replay --producer=stdout --binlog_heartbeat = true
2. Drop connection to mysql server

BinlogClient tried to reconnect only once.

```java.util.concurrent.TimeoutException: BinaryLogClient was unable to connect in 5000ms
	at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:884) ~[mysql-binlog-connector-java-0.25.3.jar:0.25.3]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.ensureReplicatorThread(BinlogConnectorReplicator.java:454) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:636) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:186) ~[classes/:?]
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:256) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:287) ~[classes/:?]
14:18:25,997 INFO  TaskManager - Stopping: com.zendesk.maxwell.schema.PositionStoreThread@345d579d
14:18:25,997 INFO  TaskManager - Stopping: com.zendesk.maxwell.bootstrap.BootstrapController@db8d926
14:18:25,998 INFO  TaskManager - Stopping: com.zendesk.maxwell.replication.BinlogConnectorReplicator@19fc5665
java.util.concurrent.TimeoutException: BinaryLogClient was unable to connect in 5000ms
	at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:884)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.ensureReplicatorThread(BinlogConnectorReplicator.java:454)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:636)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:186)
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34)
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:256)
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:287)
14:18:27,768 ERROR MaxwellContext - Exception occurred during shutdown:
java.util.concurrent.TimeoutException: Timed out trying waiting for com.zendesk.maxwell.bootstrap.BootstrapController process to stop after 1000ms.
	at com.zendesk.maxwell.util.StoppableTaskState.awaitStop(StoppableTaskState.java:46) ~[classes/:?]
	at com.zendesk.maxwell.util.RunLoopProcess.awaitStop(RunLoopProcess.java:25) ~[classes/:?]
	at com.zendesk.maxwell.util.TaskManager.stop(TaskManager.java:55) ~[classes/:?]
	at com.zendesk.maxwell.MaxwellContext.shutdown(MaxwellContext.java:192) [classes/:?]
	at com.zendesk.maxwell.MaxwellContext$1$1.run(MaxwellContext.java:223) [classes/:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
14:18:27,770 ERROR MaxwellContext - Shutdown stalled - forcefully killing maxwell process
14:18:27,770 ERROR MaxwellContext - Termination reason:
java.util.concurrent.TimeoutException: BinaryLogClient was unable to connect in 5000ms
	at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:884) ~[mysql-binlog-connector-java-0.25.3.jar:0.25.3]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.ensureReplicatorThread(BinlogConnectorReplicator.java:454) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:636) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:186) ~[classes/:?]
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:256) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:287) ~[classes/:?]
```

### PR DESCRIPTION
**Pull request by benomarek**
fixed #1768 
- when connection is dropped and binlog_heartbeat = true, BinaryLogClient will try reconnect until connection is re-established

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import com.zendesk.maxwell.scripting.Scripting;
 import com.zendesk.maxwell.util.AbstractConfig;
 import com.zendesk.maxwell.util.MaxwellOptionParser;
+import jnr.ffi.annotations.In;
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java:17]: this import seems wrong

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 2]
```diff
@@ -270,6 +272,8 @@ protected MaxwellOptionParser buildOptionParser() {
 				.withRequiredArg();
 		parser.accepts( "raft_member_id", "raft memberID.  (may also be specified in raft.xml)" )
 				.withRequiredArg();
+		parser.accepts( "client_max_reconnection_attempts", "client_max_reconnection_attempts" )
+				.withOptionalArg().ofType(Integer.class);
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java:276]: let's see... 

- let's move this to the "mysql" section of the help.  underneath `--replica_server_id` should do. 
- call it `replication_reconnection_retries`

**Comment by osheroff** [comment]
sorry let this slip by.  looks like good functionality to add, couple minor tweaks needed.

**Comment by benomarek** [comment]
@osheroff I implemented changes you mentioned above.

**Pull request merged**